doc-content/enterprise-only/openshift/offline-repo-proc.adoc

```diff
@@ -1,16 +1,54 @@
 
-. Prepare a Maven release repository to which you can write. The repository must allow read access without authentication. Your OpenShift environment must have access to this repository. You can deploy a Nexus repository manager in the OpenShift environment. For instructions about setting up Nexus on OpenShift, see https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/developer_guide/tutorials#nexus-setting-up-nexus[Setting up Nexus]. 
+* A computer that has outgoing access to the public Internet.
 
-. Complete the following actions on a computer that has an outgoing connection to the public Internet:
+.Procedure
 
+ifeval::["{context}"=="openshift-immutable"]
+. Prepare a Maven release repository to which you can write. The repository must allow read access without authentication. Your OpenShift environment must have access to this repository. You can deploy a Nexus repository manager in the OpenShift environment. For instructions about setting up Nexus on OpenShift, see https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/developer_guide/tutorials#nexus-setting-up-nexus[Setting up Nexus]. Use this repository as a mirror repository.
+ifdef::PAM[]
+If you are planning to create immutable servers from KJAR services or to deploy {CENTRAL} Monitoring,
+endif::PAM[]
+ifdef::DM[]
+If you are planning to create immutable servers from KJAR services,
+endif::DM[]
+place your services in this repository as well. You must configure this repository as the external Maven repository. You cannot configure a separate mirror repository in an immutable environment.
+endif::[]
+ifeval::["{context}"!="openshift-immutable"]
+. Prepare a Maven release repository to which you can write. The repository must allow read access without authentication. Your OpenShift environment must have access to this repository. You can deploy a Nexus repository manager in the OpenShift environment. For instructions about setting up Nexus on OpenShift, see https://access.redhat.com/documentation/en-us/openshift_container_platform/3.11/html/developer_guide/tutorials#nexus-setting-up-nexus[Setting up Nexus]. Use this repository as a separate mirror repository.
++
+Alternatively, if you use a custom external repository (for example, Nexus) for your services, you can use the same repository as a mirror repository.
+endif::[]
++
+. On the computer that has an outgoing connection to the public Internet, complete the following steps:
+.. Download the http://release-engineering.github.io/offliner/[Offliner tool].
+.. Download the `{PRODUCT_FILE}-offliner.txt` product deliverable file from the {PRODUCT_DOWNLOAD_LINK}[Software Downloads] page of the Red Hat Customer Portal.
+.. Enter the following command to download the required artifacts using the Offliner tool:
++
+[subs="attributes,verbatim,macros"]
+----
+java -jar offliner-1.3.jar -r https://maven.repository.redhat.com/ga/ -d /home/user/temp rhpam-7.3.1-offliner.txt
```

Comment:
Any specific reason for using offliner-1.3? There are newer versions and I found a bug that will be fixed in version 1.7.
Maybe we could just reference "latest-version" in the docs to avoid unnecessary updates later. Wdyt?

Reply:
All right, I did that.